crates/sui-core/src/authority/transaction_deferral.rs

```diff
@@ -90,9 +100,11 @@ pub fn transaction_deferral_within_limit(
         deferred_from_round,
     } = deferral_key
     {
-        return (future_round - deferred_from_round) <= max_deferral_rounds_for_congestion_control;
+        debug_assert!(future_round >= deferred_from_round);
```

Comment:
I think a checked_sub would be appropriate here. Since you are already adding asserts in the constructor method, the checked_sub should be safe.

The issue with adding a saturating_sub is the following: it adds a new case that has not been planned for and is almost certainly never exercised in tests. So if we ever hit this case, there is no telling what could happen.

Compare this with a checked_sub: We strongly believe the check will never fail, but we don't have to ensure that the system would work correctly in the case of overflow, because we will simply crash. Once again: better to crash than to enter an untested code path that was not anticipated in the design.